lib/ansible/modules/cloud/vmware/vmware_guest_serial_port.py

```diff
@@ -0,0 +1,469 @@
+      - ' - C(file_path) (str): Required when I(backing_type=file).
+            File path for the host file used in this backing. Fully qualified path is required, like <datastore_name>/<file_name>'
+
+extends_documentation_fragment:
+  - vmware.documentation
+
+author:
+  - Anusha Hegde (@anusha94)
+'''
+
+EXAMPLES = '''
+# Create serial ports
+- name: Create multiple serial ports with Backing type - network, pipe, device and file
+  vmware_guest_serial_port:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    validate_certs: no
+    name: "test_vm1"
+    backings:
+    - type: 'network'
+      direction: 'client'
+      service_uri: 'tcp://6000'
+      yield_on_poll: True
+    - type: 'pipe'
+      pipe_name: 'serial_pipe'
+      endpoint: 'client'
+    - type: 'device'
+      device_name: '/dev/char/serial/uart0'
+    - type: 'file'
+      file_path: '[datastore1]/file1'
+      yield_on_poll:  True
+    register: create_multiple_ports
+
+# Modify existing serial port
+- name: Modify Network backing type
+  vmware_guest_serial_port:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    name: '{{ name }}'
+    backings:
+    - type: 'network'
+      state: 'present'
+      direction: 'server'
+      service_uri: 'tcp://6000'
+  delegate_to: localhost
+
+# Remove serial port
+- name: Remove pipe backing type
+  vmware_guest_serial_port:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    name: '{{ name }}'
+    backings:
+    - type: 'pipe'
+      state: 'absent'
+  delegate_to: localhost
+
+'''
+
+RETURN = r'''
+serial_port_data:
+    description: metadata about the virtual machine's serial ports after managing them
+    returned: always
+    type: dict
+    sample: [
+        {
+          "backing_type": "network",
+          "direction": "client",
+          "service_uri": "tcp://6000"
+        },
+        {
+          "backing_type": "pipe",
+          "direction": "server",
+          "pipe_name": "serial pipe"
+        },
+    ]
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import PyVmomi, vmware_argument_spec, set_vm_power_state, wait_for_task, gather_vm_facts
+from ansible.module_utils._text import to_native
+try:
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+
+class PyVmomiHelper(PyVmomi):
+    """ This class is a helper to create easily VMware Spec for PyVmomiHelper """
+
+    def __init__(self, module):
+        super(PyVmomiHelper, self).__init__(module)
+        self.change_applied = False   # a change was applied meaning at least one task succeeded
+        self.config_spec = vim.vm.ConfigSpec()
+        self.config_spec.deviceChange = []
+        self.serial_ports = []
+
+    def check_vm_state(self, vm_obj):
```

Comment:
Hi, is this function can be simplified to 'if vm_obj.runtime.powerState != vim.VirtualMachinePowerState.poweredOff' as what used in other modules? Thanks.

Reply:
removed `gather_vm_facts` function call

Thanks for pointing out :)